### DISCUSSION TITLE
recent addition of diff2html into zowe/imperative causes webpack to fail

### DISCUSSION THREAD

**Discussion by davidkjackson54** [zowe/zowe-cli#1957] [Q&A] [answered]
I have a cli plugin for zowe that I share the classes with a VSCode Explorer extension.
After recent upgrade of the zowe/imperative version level, when I came to compile the VSCode extension using webpack, the compile now fails with this:
`ERROR in ./node_modules/wontache/module.js 1:0
Module parse failed: 'import' and 'export' may appear only with 'sourceType: module' (1:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
> import isFunction from 'underscore/modules/isFunction.js';
| import isNumber from 'underscore/modules/isNumber.js';
| import isString from 'underscore/modules/isString.js';
 @ ./node_modules/diff2html/lib-esm/hoganjs-utils.js 12:0-32 20:35-43 26:15-23
 @ ./node_modules/diff2html/lib-esm/diff2html.js 17:0-43 27:25-37
 @ ./node_modules/@zowe/imperative/lib/utilities/src/diff/DiffUtils.js 27:20-40
 @ ./node_modules/@zowe/imperative/lib/utilities/index.js 38:13-44
 @ ./node_modules/@zowe/imperative/lib/index.js 43:13-35`
 
 diff2html uses wontache under the covers and that was causing an issue with the webpack.

**Comment by davidkjackson54** [comment, marked as answer]
This solution offered by Timothy seems to have done the trick - added to webpack.config.js
```
 rules: [{
            test: /\.js$/,
            include: /wontache/,
            type: "javascript/auto"
        },
```

**Reply by t1m0thyj** [reply]
This seems to be the best solution for now. It tells Webpack to treat `wontache` as an ESM module. Found this idea here: https://github.com/webpack/webpack/issues/11597#issuecomment-705114717

For anyone else encountering this issue, another solution is to add an override to your project's package.json that pins the `diff2html` version down to "3.4.34" (or any version that does not contain the `wontache` dependency).

The first solution to add the Webpack rule is probably a better one, as setting package overrides is generally undesirable except as a last resort. Also, the override uses a version of `diff2html` that relies on 10 year old unmaintained code (see https://github.com/rtfpessoa/diff2html/issues/435).
